core/gui/src/app/hub/component/workflow/detail/hub-workflow-detail.component.ts

```diff
@@ -174,105 +154,68 @@ export class HubWorkflowDetailComponent implements AfterViewInit, OnDestroy, OnI
       this.registerReEstablishWebsocketUponWIdChange();
     }
 
-    this.registerLoadOperatorMetadata();
-
-    this.codeEditorService.vc = this.codeEditorViewRef;
+    if (this.wid) {
+      // if wid is present in the url, load it from the backend
+      this.userService
+        .userChanged()
+        .pipe(untilDestroyed(this))
+        .subscribe(() => {
+          this.loadWorkflowWithId(this.wid);
+        });
+    }
   }
 
   @HostListener("window:beforeunload")
   ngOnDestroy() {
-    if (this.workflowPersistService.isWorkflowPersistEnabled()) {
-      const workflow = this.workflowActionService.getWorkflow();
-      if (this.isLogin) {
-        this.workflowPersistService.persistWorkflow(workflow).pipe(untilDestroyed(this)).subscribe();
-      }
-    }
-
-    this.codeEditorViewRef.clear();
-    this.workflowWebsocketService.closeWebsocket();
     this.workflowActionService.clearWorkflow();
   }
 
   loadWorkflowWithId(wid: number): void {
     // disable the workspace until the workflow is fetched from the backend
     this.workflowActionService.disableWorkflowModification();
```

Comment:
do we ever enable this workflow loaded for modification? if not just put disableWorkflowModification in the constructor?